### ISSUE TITLE
When working with users from a directory, list_users and get_user return different objects.

### ISSUE THREAD

**Issue by digerata** [workos/workos-ruby#147]
Here is what we see:

__List Users:__

```ruby
irb(main):110:0> WorkOS::DirectorySync.list_users(directory: directory_id, limit: 1).data.first.username
=> "redacted@redacted.com"
irb(main):111:0> WorkOS::DirectorySync.list_users(directory: directory_id, limit: 1).data.first["username"]
Traceback (most recent call last):
        1: from (irb):111
NoMethodError (undefined method `[]' for #<WorkOS::DirectoryUser:0x00005577e10c52e8>)
```

Notice that I could take the first user in the list and then access their username with dot notation. If I tried it as a hash key then it fails.

__Get a User:__

```ruby
irb(main):106:0> WorkOS::DirectorySync.get_user(workos_user_id).username
Traceback (most recent call last):
        2: from (irb):106
        1: from (irb):106:in `rescue in irb_binding'
NoMethodError (undefined method `username' for #<Hash:0x00005577e1b665a8>)
irb(main):107:0> WorkOS::DirectorySync.get_user(workos_user_id)["username"]
=> "redacted@redacted.com"
```

Notice that I retrieve a user and I cannot use dot notation. I have to treat it as a hash.

__Expected:__

I would expect that a user from a list and a user from a get would behave the same way. Either both are accessed through hash values or both are accessed via attributes.

__Why is this an issue?__

We cannot reuse code for "syncing" a user via list and via a lookup.

**Comment by ams11** [comment]
https://github.com/workos/workos-ruby/pull/148 addresses the problem - leaving this issue open for now, until an updated gem with the fix is available.

**Comment by ams11** [comment]
This is now fixed, and a new version of the gem, containing the fix, is available 🎉 - https://rubygems.org/gems/workos/versions/2.2.1.

@digerata, thanks for reporting the issue! 🙇‍♂️ With the current changes, both end points will now always return objects (Array of objects for the list end points), so the preferred method for accessing all the attributes is the dot notation (i.e. `user.name`). For backwards compatibility, we've kept support for `[]` access, so `user["name"]` will also work (for now), but it will be deprecated in a future release, so, I'd try and move away from that style of access.
